tests/testthat/test-detect.r

```diff
@@ -49,3 +49,26 @@ test_that("str_ends works", {
   # Special typing of patterns.
   expect_true(str_ends("ab", fixed("B", ignore_case = TRUE)))
 })
+
+test_that("str_like converter works",{
+  expect_equal(like_converter("ab%"), "^ab.*$")
+  expect_equal(like_converter("ab_"), "^ab.$")
+
+  # escaping
+  expect_equal(like_converter("ab\\%"), "^ab\\%$")
+  expect_equal(like_converter("ab[%]"), "^ab[%]$")
+})
+
+test_that("str_like works", {
+  expect_true(str_like("abc", "a_c"))
```

Comment:
Once you've rewritten `str_like()` as I suggested above you'll be able to substantially simplify this test.

Reply:
Have kept only one of the general test for `str_like()` as we are testing the `like_to_regex` converter earlier. Do you think there is any further way to simplify/should there be a test to ensure it fails on any other type of regex now?

Reply:
I decided there was a slightly better way to check `pattern` so I tweaked the test too